inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -95,158 +202,312 @@ void MKLDNNPadNode::createPrimitive() {
+    }
+}
+
+void MKLDNNPadNode::execute(mkldnn::stream strm) {
     switch (padMode) {
         case CONSTANT:
-            padConstantOrEdge(srcData, dstData, srcDims, dstDims);
+            padConstant();
             break;
         case EDGE:
-            padConstantOrEdge(srcData, dstData, srcDims, dstDims, true);
+            padEdge();
             break;
         case REFLECT:
-            padReflectOrSymmetric(srcData, dstData, srcDims, dstDims);
+            padReflectOrSymmetric();
             break;
         case SYMMETRIC:
-            padReflectOrSymmetric(srcData, dstData, srcDims, dstDims, true);
+            padReflectOrSymmetric(true);
             break;
     }
 }
 
-inline size_t parallel_init(size_t start, size_t size, std::vector<size_t> &counters, std::vector<size_t> &dims) {
-    for (int j = size - 1; j >= 0; j--) {
-        counters[j] = start % dims[j];
+static inline size_t parallel_init(size_t start, size_t nDims, const SizeVector& dims, SizeVector& indexes) {
+    for (int j = nDims - 1; j >= 0; j--) {
+        indexes[j] = start % dims[j];
         start = start / dims[j];
     }
     return start;
 }
 
-inline void parallel_step(size_t size, std::vector<size_t> &counters, std::vector<size_t> &dims) {
-    for (int j = size - 1; j >= 0; j--) {
-        counters[j] = (counters[j] + 1) % dims[j];
-        if (counters[j] != 0)
+static inline void parallel_step(size_t nDims, const SizeVector& dims, SizeVector& indexes) {
+    for (int j = nDims - 1; j >= 0; j--) {
+        indexes[j] = (indexes[j] + 1) % dims[j];
+        if (indexes[j] != 0)
             return;
     }
 }
 
-void MKLDNNPadNode::padConstantOrEdge(const float* srcData, float* dstData, SizeVector srcDims, SizeVector dstDims,
-        const bool isEdge) {
-    size_t dimsSize_1 = dstDims.size() - 1;
-    size_t inputSV = dstDims[dimsSize_1];
-    size_t workAmountDst = getWorkAmountDst();
+void MKLDNNPadNode::padConstant() {
+    if (padValue == 0) {
+        padConstantZero();
+        return;
+    }
+
+    InferenceEngine::Precision precision = this->getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].desc.getPrecision();
+    OV_SWITCH(MKLDNNPlugin, PadConstantEmitter, this, precision,
+              OV_CASE(InferenceEngine::Precision::FP32, float),
+              OV_CASE(InferenceEngine::Precision::I64, int64_t),
+              OV_CASE(InferenceEngine::Precision::U64, uint64_t),
+              OV_CASE(InferenceEngine::Precision::I32, int32_t),
+              OV_CASE(InferenceEngine::Precision::BF16, bfloat16_t),
+              OV_CASE(InferenceEngine::Precision::I16, int16_t),
+              OV_CASE(InferenceEngine::Precision::U16, uint16_t),
+              OV_CASE(InferenceEngine::Precision::I8, int8_t),
+              OV_CASE(InferenceEngine::Precision::U8, uint8_t));
+}
+
+template<typename T>
+void MKLDNNPadNode::padConstantCommon() {
+    const T *srcData = getParentEdgeAt(0)->getBlob()->cbuffer().as<const T*>() +
```

Comment:
getDataPtr?

Reply:
We can't use this method, because for better perfomance when padValue != 0 we should use `std::fill_n` which doesn't support `uint8_t*` in `getDataPtr`. When you want to use this method, you must use `cpu_memcpy`, not `std::fill_n`. But its slower. So I preferred `std::fill_n` and `getBlob()->cbuffer()`

Reply:
reinterpret_cast<T*>(getDataPtr(...))?